java/vector/src/main/java/org/apache/arrow/vector/ipc/ArrowFileWriter.java

```diff
@@ -116,4 +118,63 @@ public List<ArrowBlock> getRecordBlocks() {
   public List<ArrowBlock> getDictionaryBlocks() {
     return dictionaryBlocks;
   }
+
+  /**
+   * Class to configure builder pattern for ArrowWriter that writes out a Arrow files.
+   */
+  public static class Builder extends ArrowWriter.Builder<Builder> {
+    private DictionaryProvider provider = null;
+    private IpcOption option = IpcOption.DEFAULT;
+    private Map<String, String> metaData = Collections.emptyMap();
+    private int level = -1;
+
+    public Builder(VectorSchemaRoot root, WritableByteChannel out) {
+      super(root, out);
+    }
+
+    @Override
+    public ArrowFileWriter build() {
+      ArrowFileWriter writer = new ArrowFileWriter(this);
+      writer.initialize();
+      return writer;
+    }
+
+    @Override
+    protected Builder self() {
+      return this;
+    }
+
+    public Builder setProvider(DictionaryProvider provider) {
+      this.provider = provider;
+      return this;
+    }
+
+    public Builder setOption(IpcOption option) {
+      this.option = option;
+      return this;
+    }
+
+    public Builder setMetaData(Map<String, String> metaData) {
+      this.metaData = metaData;
+      return this;
+    }
+
+    public Builder setLevel(int level) {
+      this.level = level;
+      return this;
+    }
+
+    @Override
+    public Builder setCodec(CompressionCodec codec) {
+      return super.setCodec(codec);
+    }
+  }
+
+  private ArrowFileWriter(Builder builder) {
```

Comment:
I would rather you call one of the existing constructors instead of tying the builder/class so closely together, and that will avoid needing the separate initialize() call